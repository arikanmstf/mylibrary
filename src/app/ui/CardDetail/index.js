// @flow

export { default } from './CardDetail';
