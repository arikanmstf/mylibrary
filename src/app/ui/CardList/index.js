// @flow

export { default } from './CardList';
