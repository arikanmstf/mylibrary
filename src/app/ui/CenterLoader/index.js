// @flow

export { default } from './CenterLoader';
