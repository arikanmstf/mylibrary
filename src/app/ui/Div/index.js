// @flow

export { default } from './Div';
