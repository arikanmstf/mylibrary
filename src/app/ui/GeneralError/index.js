// @flow

export { default } from './GeneralError';
