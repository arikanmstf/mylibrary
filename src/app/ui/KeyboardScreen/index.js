// @flow

export { default } from './KeyboardScreen';
