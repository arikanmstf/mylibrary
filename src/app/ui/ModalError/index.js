// @flow

export { default } from './ModalError';
