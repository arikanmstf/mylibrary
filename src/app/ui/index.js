export { default as Text } from './Text';
export { default as Button } from './Button';
export { default as Screen } from './Screen';
export { default as Image } from './Image';
export { default as TextField } from './forms/TextField';
export { default as Div } from './Div';
export { default as Col } from './Col';
export { default as Form } from './Form';
export { default as ModalLoader } from './ModalLoader';
export { default as Header } from './Header';
export { default as Page } from './Page';
export { default as CardList } from './CardList';
export { default as CardDetail } from './CardDetail';
export { default as CenterLoader } from './CenterLoader';
export { default as KeyboardScreen } from './KeyboardScreen';
export { default as GeneralError } from './GeneralError';
export { default as RowList } from './RowList';
export { default as RowDetail } from './RowDetail';
export { default as Routes } from './Routes';
export { default as Link } from './Link';
export { default as SideNavigation } from './SideNavigation';
export { default as Modal } from './Modal';
export { default as ModalError } from './ModalError';
