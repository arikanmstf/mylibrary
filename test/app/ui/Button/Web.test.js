import React from 'react';
import { shallow } from 'enzyme';
import { Button } from 'ui';

describe('test/app/ui/Button/Web.test.js', () => {
  it('Render: props: { text }', () => {
    const props = {
      text: 'dummyText',
    };

    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render: props: { primary }', () => {
    const props = {
      text: 'dummyText',
      primary: true,
    };

    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render: props: { raised }', () => {
    const props = {
      text: 'dummyText',
      raised: true,
    };

    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render: props: { onPress }', () => {
    const props = {
      text: 'dummyText',
      onPress: jest.fn(() => 'returnValue'),
    };

    const wrapper = shallow(<Button {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
