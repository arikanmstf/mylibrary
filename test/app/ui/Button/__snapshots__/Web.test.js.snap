// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/ui/Button/Web.test.js Render: props: { onPress } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onPress={[MockFunction]}
    text="dummyText"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "dummyText",
      "color": undefined,
      "onPress": [MockFunction],
      "style": Object {
        "width": "100%",
      },
      "variant": undefined,
    },
    "ref": null,
    "rendered": "dummyText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "dummyText",
        "color": undefined,
        "onPress": [MockFunction],
        "style": Object {
          "width": "100%",
        },
        "variant": undefined,
      },
      "ref": null,
      "rendered": "dummyText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`test/app/ui/Button/Web.test.js Render: props: { primary } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    primary={true}
    text="dummyText"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "dummyText",
      "color": "primary",
      "style": Object {
        "width": "100%",
      },
      "variant": undefined,
    },
    "ref": null,
    "rendered": "dummyText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "dummyText",
        "color": "primary",
        "style": Object {
          "width": "100%",
        },
        "variant": undefined,
      },
      "ref": null,
      "rendered": "dummyText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`test/app/ui/Button/Web.test.js Render: props: { raised } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    raised={true}
    text="dummyText"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "dummyText",
      "color": undefined,
      "style": Object {
        "width": "100%",
      },
      "variant": "contained",
    },
    "ref": null,
    "rendered": "dummyText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "dummyText",
        "color": undefined,
        "style": Object {
          "width": "100%",
        },
        "variant": "contained",
      },
      "ref": null,
      "rendered": "dummyText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`test/app/ui/Button/Web.test.js Render: props: { text } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="dummyText"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "dummyText",
      "color": undefined,
      "style": Object {
        "width": "100%",
      },
      "variant": undefined,
    },
    "ref": null,
    "rendered": "dummyText",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "dummyText",
        "color": undefined,
        "style": Object {
          "width": "100%",
        },
        "variant": undefined,
      },
      "ref": null,
      "rendered": "dummyText",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
